### DISCUSSION TITLE
Getting Error `You might be breaking the Rules of Hooks` when following the "slices-pattern" example

### DISCUSSION THREAD

**Discussion by HerrZatacke** [pmndrs/zustand#1783] [Q&A] [answered]
I'm trying to follow this example: https://github.com/pmndrs/zustand/blob/e71e313bd715af92bf235c4240fb9267ddaa5b4f/docs/guides/slices-pattern.md

I get this error:
```
Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://reactjs.org/link/invalid-hook-call for tips about how to debug and fix this problem.
```

It seems that in the example the calls to `createBearSlice` and `createFishSlice` in the third example are causing the issue.

I'm using
```
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "zustand": "^4.3.7"
```
Any advice on what I'm doing wrong?

**Comment by dbritto-dev** [comment]
@HerrZatacke thanks for reporting it. I'll look into that. Would you mind providing us a codesandbox example of that?.

**Reply by HerrZatacke** [reply]
> @HerrZatacke thanks for reporting it. I'll look into that. Would you mind providing us a codesandbox example of that?.

I pushed a demo to my affected repository: (Branch `bug/zustand`)
https://github.com/HerrZatacke/projectkick/tree/bug/zustand

The relevant stores are in [`src/javascript/app/stores`](https://github.com/HerrZatacke/projectkick/tree/bug/zustand/src/javascript/app/stores)
The affected component is in [`src/javascript/app/components/App/index.jsx`](https://github.com/HerrZatacke/projectkick/blob/bug/zustand/src/javascript/app/components/App/index.jsx)

**Reply by dbritto-dev** [reply]
@HerrZatacke would you mind updating your project to use `vitejs`? seems you are not using even `CRA` and you custom webpack config seems wrong. I recommend you to start a new fresh project using `vitejs`.

**Reply by HerrZatacke** [reply]
My project has been running fin the last ~5 years.

But to make it easier for you I reproduced the bug in a fresh repository created with CRA:
https://github.com/HerrZatacke/bug-demo/

**Reply by dbritto-dev** [reply, marked as answer]
> My project has been running fin the last ~5 years.
> 
> But to make it easier for you I reproduced the bug in a fresh repository created with CRA:
> https://github.com/HerrZatacke/bug-demo/
> 

It's time to upgrade. BTW, thanks for the new repo I was able to run it and I found the issue. Basically, when you are working with slices `createExampleStore` and `createApiExampleStore` should be just functions since those are slices of a store those aren't stores by itself. I would rename them to `createExampleSlice` and `createApiExampleSlice` to match their current behavior.

What happen if you use `create`? Well, you are creating individual stores (custom hooks) and when you call them on `boundStore` since is not a component you are breaking the rules of hooks.

**Solution**

For slices you should not use `create`

```ts
(set) => ({ })
```

> **Note:** here's the codesandbox https://codesandbox.io/p/github/dbritto-dev/bug-demo/draft/tender-beaver?workspaceId=8aa24f99-7df1-4511-addd-431f096678d9&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A1%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A1%7D%5D

**Reply by HerrZatacke** [reply]
Ah. You're right. I missed that. Thanks a lot for your help!!
